test/should.js

```diff
@@ -1407,6 +1407,16 @@ describe('should', function() {
 
     ({ 'foo.bar[]': 'baz'}).should.have.nested.property('foo\\.bar\\[\\]');
 
+    ({a:1}).should.have.nested.property('a');
+
+    (function () {
+      ({a:1}).should.have.nested.property('{a:1}');
+    }).should.not.throw('the argument to property must be a string when using nested syntax');
```

Comment:
Let's remove this test. We already test that `.nested.property` works properly elsewhere in the suite. Adding this test with ".should.not.throw` doesn't add any value.